opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchRequestBuilder.java

```diff
@@ -173,14 +175,33 @@ public void pushDownLimit(Integer limit, Integer offset) {
    * Add highlight to DSL requests.
    * @param field name of the field to highlight
    */
-  public void pushDownHighlight(String field) {
+  public void pushDownHighlight(String field, Map<String, Literal> arguments) {
+    String unquotedField = StringUtils.unquoteText(field);
     if (sourceBuilder.highlighter() != null) {
-      sourceBuilder.highlighter().field(StringUtils.unquoteText(field));
+      // OS does not allow duplicates of highlight fields
+      if (sourceBuilder.highlighter().fields().stream()
+          .anyMatch(f -> f.name().equals(unquotedField))) {
+        throw new SemanticCheckException(String.format(
+            "Duplicate field %s in highlight", field));
+      }
+
+      sourceBuilder.highlighter().field(unquotedField);
     } else {
       HighlightBuilder highlightBuilder =
-          new HighlightBuilder().field(StringUtils.unquoteText(field));
+          new HighlightBuilder().field(unquotedField);
       sourceBuilder.highlighter(highlightBuilder);
     }
+
+    // lastFieldIndex denotes previously set highlighter with field parameter
+    int lastFieldIndex = sourceBuilder.highlighter().fields().size() - 1;
+    if (arguments.containsKey("pre_tags")) {
+      sourceBuilder.highlighter().fields().get(lastFieldIndex)
+          .preTags(arguments.get("pre_tags").toString());
+    }
+    if (arguments.containsKey("post_tags")) {
+      sourceBuilder.highlighter().fields().get(lastFieldIndex)
+          .postTags(arguments.get("post_tags").toString());
+    }
```

Comment:
Do you have a test which covers such case? A few `highlight`'s with different tags

Reply:
I have SQL and PPL tests covering these:
[ppl/HighlightFunctionIT.](https://github.com/Bit-Quill/opensearch-project-sql/blob/d1abeb20522e81ab8adc9c20ea00770159f115f8/integ-test/src/test/java/org/opensearch/sql/ppl/HighlightFunctionIT.java#L70-L95)
[sql/sql/HighlightFunctionIT](https://github.com/Bit-Quill/opensearch-project-sql/blob/d1abeb20522e81ab8adc9c20ea00770159f115f8/integ-test/src/test/java/org/opensearch/sql/sql/HighlightFunctionIT.java#L47-L64)